### PR TITLE
[mlir][vector] Add more tests for ConvertVectorToLLVM (4/n)

### PR DESCRIPTION
**Pull request by banach-space**
Adds tests with scalable vectors for the Vector-To-LLVM conversion pass.
Covers the following Ops:
  * vector.insertelement
  * vector.insert

I have also renamed some function names from `@insert_element{}` to
`@insertelement{}` - that's to make a clearer distinction between
tests for `vector.insertelement` (tested by `@insertelement{}`) and
`vector.insert` (tested by `@insert_element{}`).

### CODE REVIEW

**Review by Azooz1988 — APPROVED** [review]

File: mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir [hunk 1]
```diff
@@ -1258,18 +1258,30 @@ func.func @insert_element_0d(%a: f32, %b: vector<f32>) -> vector<f32> {
...
+// CHECK-LABEL: @insertelement_scalable(
+// CHECK-SAME: %[[A:.*]]: f32,
+// CHECK-SAME: %[[B:.*]]: vector<[4]xf32>)
+//       CHECK:   %[[c:.*]] = arith.constant 3 : i32
+//       CHECK:   %[[x:.*]] = llvm.insertelement %[[A]], %[[B]][%[[c]] : i32] : vector<[4]xf32>
+//       CHECK:   return %[[x]] : vector<[4]xf32>
+
 // -----
 
 func.func @insert_element_index(%arg0: f32, %arg1: vector<4xf32>) -> vector<4xf32> {
```
Comment (nujaa) [inline mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:1287]:
```suggestion
func.func @insertelement_index(%arg0: f32, %arg1: vector<4xf32>) -> vector<4xf32> {
```

**Review by nujaa — COMMENTED** [review]
Hi, I got a note for maybe another MR, 

> tests for vector.insertelement (tested by @insertelement{}) and
vector.insert (tested by @insert_element{}).
 Do you think we should rename tests on `vector.insert` to not have a name like `insert_element*` for more clarity ? like `@insert_vec_1d_into_vec_3d`.

We could imagine for example `@insert_element_into_vec_3d` to be renamed into `@insert_scalar_into_vec_3d`.  
Also, for the `insert_element_with_value*`  tests. `with_value` means that the index is dynamic, I suppose ?
How about ` @insert_element_with_value_1d` => ` @insert_scalar_into_vec_1d_dynamic_index`

File: mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir (hunk 1 above)
Comment (banach-space) [reply mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir]: Good catch, thanks!

**Comment by banach-space** [comment]
Ha, good points!

> Do you think we should rename tests on `vector.insert` to not have a name like `insert_element*` for more clarity ? like `@insert_vec_1d_into_vec_3d`.

+1

> We could imagine for example `@insert_element_into_vec_3d` to be renamed into `@insert_scalar_into_vec_3d`.

+1 

IMO, this would be most natural naming scheme: `@op-name_some_unique_features`. 

Basically, what you proposed there ^^^ :)

> Also, for the `insert_element_with_value*` tests. `with_value` means that the index is dynamic, I suppose ? How about ` @insert_element_with_value_1d` => ` @insert_scalar_into_vec_1d_dynamic_index`

I was thinking exactly the same :)

I am about to disappear for a week and would love to land this as is if that's OK with you. Would you be willing to send a PR to rename the tests? 

Thanks for taking a look 🙏🏻

File: mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir [hunk 2]
```diff
@@ -1284,7 +1284,7 @@ func.func @insertelement_scalable(%arg0: f32, %arg1: vector<[4]xf32>) -> vector<
 
 // -----
 
-func.func @insert_element_index(%arg0: f32, %arg1: vector<4xf32>) -> vector<4xf32> {
+func.func @insertelement_index(%arg0: f32, %arg1: vector<4xf32>) -> vector<4xf32> {
```
Comment (nujaa) [inline mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:1287]: You didnt update the CHECK :|

**Review by nujaa — COMMENTED** [review]
CI won't pass I think.
Otherwise LGTM.

**Comment by nujaa** [comment]
> I was thinking exactly the same :)

Great minds think alike.

> I am about to disappear for a week and would love to land this as is if that's OK with you. Would you be willing to send a PR to rename the tests?

Sure, with pleasure. Since I suspect your CI will fail with your last fixup, I let you update it. In case at the end of the day you haven't, I'll suspect you are already gone and will fix and merge.

**Review by nujaa — APPROVED** [review]
LGTM.

**Pull request merged**